### DISCUSSION TITLE
The test images I generated contained meaningless content.

### DISCUSSION THREAD

**Discussion by Aloereed** [Amblyopius/Stable-Diffusion-ONNX-FP16#13] [Q&A]
Intel A770 sd-2-1base
I tried to use the GUI to generate other content, but the images generated were similar to this.
![testpicture-sd2_1base-fp16_512_seed0](https://user-images.githubusercontent.com/58137529/219830973-bf1ee2f4-1fb7-4328-a97a-54162ccfcc7e.png)

Is it normal?

**Comment by Amblyopius** [comment]
Hi, not expected at all. As we're using DirectML everything should work on any DirectX 12.1 card, as long as the DirectX driver is not bugged.

You should have output from the test in FP32 and FP16. Are both useless content? If both are useless, can you remake the FP32 one with the following line:
python conv_sd_to_onnx.py --model_path "stabilityai/stable-diffusion-2-1-base" --output_path "./model/sd2_1base-fp32-notune" --notune
and let me know if that changes something. If the test image from that (using test-txt2img.py) looks fine you can test fp16 by converting it like this:
python conv_sd_to_onnx.py --model_path "stabilityai/stable-diffusion-2-1-base" --output_path "./model/sd2_1base-fp16-notune" --fp16 --notune

If the FP32 one was fine but FP16 not then there's only an issue with FP16 and you should stick to using FP32. It would be best to use at least '--attention-slicing auto' when creating the models as FP32 uses more VRAM (but you have 16GB so it would still work). And there's no need to try with --notune if the FP32 was already fine.

You can always join us on the nod.ai discord server on the #onnxdiffusers channel. There are also plans for nod.ai to add A770 support to SHARK (although the Vulkan drivers for A770 are unfortunately enough also not where one would want them to be).

Your other option is to run with [Intel Extension for Pytorch](https://github.com/intel/intel-extension-for-pytorch) but it will require code changes and I don't know how comfortable you're with doing that.

**Comment by Aloereed** [comment]
I've deleted the sd2-1-base cache. But I tried andite/anything-4.0 and the results were similar, even with notune and fp32.

![anything-4-fp32-notune_512_seed0](https://user-images.githubusercontent.com/58137529/219870878-c133a43c-b287-4189-afbf-373d4a754d8d.png)

 I also tried Intel extension([rahulunair](https://github.com/rahulunair)
 /
[stable_diffusion_arc](https://github.com/rahulunair/stable_diffusion_arc)) on WSL2, But it still doesn't produce a meaningful picture.

![And o](https://user-images.githubusercontent.com/58137529/219870891-4d7bdaaa-ff11-48d8-94da-0bbf3cce44ae.png)

So I began to suspect a device driver problem.

**Reply by Amblyopius** [reply]
I unfortunately have no Arc hardware to test on but there's a Beta driver from 16th of Feb and a WHQL one from 1st of Feb. Both recent enough to be worth a try. The beta is only 2 days old so worth a shot if you did not update recently and if that's the one you're on then maybe try the WHQL as they may have released a lemon on the 16th.

**Reply by Aloereed** [reply]
Upgrading the driver did not help. 

And is it normal to have this during converting to onnx? 
TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!

**Reply by Amblyopius** [reply]
Yes, those warnings are normal. They just mean that in some areas we can only use the default in ONNX. As a result we can't support some specific things you can do as an option. For example Guidance Scale works differently at 1 or below. It does not work on ONNX when you do that as that's one of the things where things have been treated as constants. A lot of these things never happen, the guidance scale one is one of the few someone may try at some point.

**Reply by Aloereed** [reply]
I used lshqqytiger/stable-diffusion-webui-directml successfully, so the driver should be fine. 
I will try this project a few more times afterwards.

**Reply by Amblyopius** [reply]
That implementation uses torch DirectML which unfortunately is still extremely slow and buggy when compared to ONNX Runtime. But apparently it works better with A770 at the moment. As that installation is essentially the most used UI (A1111 web UI), having it working will give you far more features than you'd get with ONNX Runtime. You'll just have to live with the performance or rather lack thereof.

I have too many projects that break with torch DirectML but fingers crossed that enough is fixed in the next version. Their release cycle is very patchy but they did one in December and one in January so maybe we've a February one coming too.

**Reply by Aloereed** [reply]
Yes, I hope that both Torch-DirectML and ONNX will be even better in the near future on the Intel Arc platform. Right now the Torch-DirectML WebUI is really slow.
